Manjula Mishra [DS-Unit-4-Sprint-3-Neural-Networks]

manjulamishra opened the pull request:
Not started yet

manjulamishra committed:
Assignment initial commit

Not started yet

manjulamishra committed:
Merge pull request #1 from LambdaSchool/master

adds backpropagation content

manjulamishra committed:
half finished

manjulamishra committed:
Finished homework

But still have lots of confusions

manjulamishra committed:
Mon-Tuesday assignments

unfinished

kovacova reviewed (commented):
Well done Manjula, that's a lot of good progress work with excellent visualizations! Good to see both MNIST and Iris attempts. Nice work!

T.

manjulamishra committed:
SC finished

manjulamishra committed:
Some changes

Tried doing 1st problem also how we did in the class. Doesn't seem to predict AND gate the way first one does.

kovacova reviewed (commented):
## Sprint Challenge Code Review
### Week 17 - Neural Networks
---
#### Part 1 — Definitions
You correctly defined all terms, and your explanations show a strong understanding and mastery of these fundamental Neural Network terms.

---
#### Part 2 — Perceptron Implementation
You implemented a perceptron class, trained it on the training data and updated weights appropriately until the output from the network is very close to the expected training data output (y).

---
#### Part 3 — MLP Implementation
Your MLP implementation correctly initializes weights, takes in inputs, combines inputs and weights as a weighted sum, activates the weighted sum, and passes the activated values onto the next layer. Backpropagation implementation appropriately updates weights in reverse order and leads to performance improvements in the model. Model's loss should decrease and accuracy improve over multiple iterations.

---
#### Part 4 — Keras MLP and Hyperparameter Tuning
Your Keras implementation appropriately uses a sequential model with dense layers, and appropriate architecture with at least one hidden layer, activation functions, appropriate loss function, and accuracy metric. Model's verbose output shows significant improvement from first to last epoch during training. You showed your work in hyperparameter tuning two features and reported the CV accuracy of each parameter combination that is being tested.

<details><summary>Well done, Manjula!</summary>

<img src="https://www.nola.com/resizer/uY8mRDyl8SJ3pNBYMu3P2THjnJc=/600x0/arc-anglerfish-arc2-prod-advancelocal.s3.amazonaws.com/public/F6ASCVG4ZBBEDDMKDIFKWAYYOU.jpg" width="250"></details>
